File: app/src/main/java/com/example/student_attendance_ms/main/schedule/detail/EventDetailAdapter.java
package com.example.student_attendance_ms.main.schedule.detail;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.student_attendance_ms.R;
import com.example.student_attendance_ms.network.model.EventMember;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class EventDetailAdapter extends RecyclerView.Adapter<EventDetailAdapter.UserEntriesViewHolder> {

    private List<EventMember> userEntries;

    public EventDetailAdapter(List<EventMember> userEntries){
        this.userEntries = userEntries;
    }

    @NonNull
    @Override
    public UserEntriesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(
                R.layout.list_event_members,
                parent,
                false
        );
        return new UserEntriesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UserEntriesViewHolder holder, int position) {
        holder.bind(userEntries.get(position));
    }

    @Override
    public int getItemCount() {
        return userEntries.size();
    }

    static class UserEntriesViewHolder extends RecyclerView.ViewHolder{

        final CircleImageView userImage;
        final TextView userName;

        UserEntriesViewHolder(@NonNull View itemView) {
            super(itemView);
            userImage = itemView.findViewById(R.id.profile_image);
            userName = itemView.findViewById(R.id.person_name);
        }

        void bind(EventMember userEntries){
            userImage.setImageResource(userEntries.getUserImage());
            userName.setText(userEntries.getUserName());
        }
    }

}

File: app/src/main/java/com/example/student_attendance_ms/main/schedule/detail/EventDetailAdapter.kt
package com.example.student_attendance_ms.main.schedule.detail;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView;
import com.example.student_attendance_ms.databinding.ListEventMembersBinding
import com.example.student_attendance_ms.network.model.EventMember;

class EventDetailAdapter : ListAdapter<EventMember, EventDetailAdapter.EventDetailViewHolder>(DiffCallback) {

    class EventDetailViewHolder(
            private val binding: ListEventMembersBinding
    ) : RecyclerView.ViewHolder(binding.root) {
        fun bind(eventMember: EventMember) {
            binding.eventMember = eventMember
            binding.executePendingBindings()
        }
    }

    override fun onCreateViewHolder(
            parent: ViewGroup,
            viewType: Int
    ): EventDetailViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = ListEventMembersBinding.inflate(
                layoutInflater, parent, false
        )
        return EventDetailViewHolder(binding)
    }

    override fun onBindViewHolder(
            holder: EventDetailViewHolder,
            position: Int
    ) = holder.bind(getItem(position))


    companion object DiffCallback : DiffUtil.ItemCallback<EventMember>() {
        override fun areItemsTheSame(oldItem: EventMember, newItem: EventMember): Boolean {
            return oldItem === newItem
        }

        override fun areContentsTheSame(oldItem: EventMember, newItem: EventMember): Boolean {
            return oldItem.memberId == newItem.memberId
        }
    }
}